### PR TITLE
#128 [LAD] LAD should restart OMI if it crashes

### PR DESCRIPTION
**Pull request by hosungsmsft**
@LiweiPeng & @jasonzio : Please review this PR. Thanks.

### CODE REVIEW

File: Diagnostic/diagnostic.py [hunk 1]
```diff
@@ -648,14 +649,27 @@ def start_mdsd():
...
                     break
+
+                # mdsd is now up for at least 30 seconds.
+
+                # Issue #128 LAD should restart OMI if it crashes
+                omi_up_and_running = RunGetOutput("/opt/omi/bin/omicli noop")[0] is 0
+                if not omi_up_and_running:
+                    hutil.error("OMI noop query failed. OMI crash suspected. Restarting OMI and send SIGHUP to mdsd after 5 seconds.")
+                    omi_restart_msg = RunGetOutput("/opt/omi/bin/service_control restart")[1]
+                    hutil.error("OMI restart result: " + omi_restart_msg)
```
Comment (jasonzio) [inline Diagnostic/diagnostic.py]: You should make sure OMI comes up and stays up before you send SIGHUP to mdsd. If you can't get OMI to come up and stay up, I would suggest logging something to syslog, although I'm not sure what facility to recommend.

Comment (hosungsmsft) [inline Diagnostic/diagnostic.py]: Makes sense. Will add the code to check that as the condition of mdsd.send_signal(signal.SIGHUP). Thx.

**Comment by jasonzio** [comment]
LGTM

**Comment by hosungsmsft** [comment]
Thanks Jason. Merged and closing this PR.

**Pull request merged**